src/test/java/org/opensearch/knn/index/LuceneEngineIT.java

```diff
@@ -338,6 +338,26 @@ public void testQuery_filterWithNonLuceneEngine() throws Exception {
         );
     }
 
+    public void testIndexReopening() throws Exception {
+        createKnnIndexMappingWithLuceneEngine(DIMENSION, SpaceType.L2);
+
+        for (int j = 0; j < TEST_INDEX_VECTORS.length; j++) {
+            addKnnDoc(INDEX_NAME, Integer.toString(j + 1), FIELD_NAME, TEST_INDEX_VECTORS[j]);
+        }
+
+        final float[] searchVector = TEST_QUERY_VECTORS[0];
+        int k = 1;
```

Comment:
Is `k = 1` appropriate value to detect an issue? Does this test fail without your fix?

Reply:
exact k actually doesn't matter, purpose is to validate that index in operable and search queries are returning results and not failing with error.

Reply:
Got it. Then, can we just execute the query and don't validate the k value then? It would be nice if other see the code and understand that we are just expecting no exception here.

Reply:
We'll not have an exception, as it's a rest call and the closes analogy is status code. It's not exactly same as checking for result. Checking for number of vectors is already a relaxed validation, as we're typically checking if returned vector is actually a closest neighbor by checking distance and score https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/index/LuceneEngineIT.java#L391.

Reply:
I think at least we need to validate not just the number of results but if the response is same before closing the index and after reopening the index. Also having k as a random value would be nice.

Reply:
yeah, we can compare if results are same between two calls, also agree on random k, let me make the change